### PR TITLE
Address LGTM alerts

### PR DESCRIPTION
**Pull request by jwodder**
Some alerts are false positives and thus are not addressed by this PR.

### CODE REVIEW

File: dandi/_version.py [hunk 1]
```diff
@@ -504,7 +504,7 @@ def get_versions():
         # versionfile_source is the relative path from the top of the source
         # tree (where the .git directory might live) to this file. Invert
         # this to find the root from __file__.
-        for i in cfg.versionfile_source.split('/'):
+        for _ in cfg.versionfile_source.split('/'):
```
Comment (yarikoptic) [inline dandi/_version.py:507]: this is versioneer one, can we just skip it for LGTM?

meanwhile checked -- current master in versioneer already has this:
```
$> git grep -A1 'in cfg.versionfile_source.split'
src/git/long_get_versions.py:        for _ in cfg.versionfile_source.split('/'):
src/git/long_get_versions.py-            root = os.path.dirname(root)
```
so in principle we could just proceed, but still I think we should just skip this file

File: tools/validate-api-against-girder.py [hunk 2]
```diff
@@ -1,16 +1,12 @@
 #!/usr/bin/env python3
 import click
-import requests
 
 from dandi.dandiapi import DandiAPIClient
-from dandi.dandiset import APIDandiset
 from dandi.girder import GirderCli
```
Comment (yarikoptic) [inline tools/validate-api-against-girder.py:5]: let's kill this file altogether -- girder is gone gone

File: dandi/_version.py (hunk 1 above)
Comment (jwodder) [reply dandi/_version.py:507]: I don't know how ignoring alerts works in LGTM, so I can't answer that.

File: tools/validate-api-against-girder.py (hunk 2 above)
Comment (jwodder) [reply tools/validate-api-against-girder.py:5]: Should `tools/instantiate-dandisets.py` (which also uses Girder) be deleted as well?

Comment (yarikoptic) [reply tools/validate-api-against-girder.py:5]: sure, why to drag it now

File: dandi/_version.py (hunk 1 above)
Comment (yarikoptic) [reply dandi/_version.py:507]: google has all the answers: https://lgtm.com/help/lgtm/customizing-file-classification

File: dandi/due.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def _donothing_func(*args, **kwargs):
 
 
 try:
-    from duecredit import due, BibTeX, Doi, Url, Text
+    from duecredit import due, Doi  # lgtm [py/unused-import]
```
Comment (yarikoptic) [inline dandi/due.py:58]: Let's skip this file as well from lgtm checks - it is a stub which comes from duecredit and better be left unmodified

Comment (jwodder) [reply dandi/due.py:58]: Do you want me to revert the changes to our local `due.py`?

Comment (yarikoptic) [reply dandi/due.py:58]: yes, to due.py and `_version.py` and list them to be excluded in to be added lgtm.yml per https://lgtm.com/help/lgtm/customizing-file-classification

Comment (yarikoptic) [reply dandi/due.py:58]: oh I see you added already lgtm.yml -- great

Comment (jwodder) [reply dandi/due.py]: Done.

**Comment by yarikoptic** [comment]
Thank you @jwodder !  Let's proceed.

I feel we might like to pacify this LGTM bot to not add comments -- adding to the checks list is IMHO sufficient.

**Pull request merged**